### PR TITLE
More robust exn handler

### PR DESCRIPTION
**Pull request by vext01**
VM testing has raised a couple of other issues.

See individual commit messages.

OK?

### CODE REVIEW

File: krun/scheduler.py [hunk 1]
```diff
@@ -367,16 +367,16 @@ def run(self):
         job = ExecutionJob(self, vm, self.config.VMS[vm], bench, variant,
                            self.config.BENCHMARKS[bench])
 
+        # Run the user's pre-process-execution commands We can't put an ETA
```
Comment (ltratt) [inline krun/scheduler.py]: Missing "." before "we".

**Comment by ltratt** [comment]
The first commit opens up a nasty race where *part* of the pre hooks fail, but then none of the post hooks are run. For example, the network might be brought down, then something else goes wrong in the pre hooks; with this commit, the post hook networking-bringer-uper won't be run. This was something I was very conscious of when I suggested the layout of the try/finally, so I'd need to be convinced that it should be changed. I do think that we should explicitly document that the post hooks are run even if some of the pre hooks failed though.

**Comment by snim2** [comment]
The second commit looks sensible to me, the first needs some discussion.

**Comment by vext01** [comment]
The network down command is last in the pre-hook, so we are grand. Right?

**Comment by vext01** [comment]
It's not just about the network. Point taken.

**Comment by snim2** [comment]
I think if we are careful about the network and `| true` in the right places then this looks OK. Let's wait for @ltratt before we squash though.

**Comment by vext01** [comment]
@snim2 laurie is right.

Suppose we have pre-hooks:
```
rcctl stop postfix
rcctl stop cron
```

and post hooks:
```
rcctl start postfix
rcctl start cron
```

And `rcctl stop cron` fails. Postfix would not be brought up.

My last push reverts the old behaviour and explains the placement in a comment.

**Comment by vext01** [comment]
If it's OK, I would like to revise some orthogonal comments in nearby code. Can I add this change here?

**Comment by snim2** [comment]
Sure. BTW the comment about dumping JSON on lines 412-413 of `scheduler.py` does not seem to apply to the line of code beneath it.

**Comment by vext01** [comment]
That's the very line of code I want to change ;)

**Comment by vext01** [comment]
Here we go. OK to squash?

**Comment by snim2** [comment]
OK, lgtm

**Comment by vext01** [comment]
splat

**Pull request merged**